sherpa/astro/data.py

```diff
@@ -3453,7 +3558,10 @@ def ignore_bad(self):
         # self.quality_filter used for pre-grouping filter
         self.quality_filter = qual_flags
 
-    def _dynamic_group(self, group_func, *args, **kwargs):
+    def _dynamic_group(self,
+                       # TODO: remove callable support
```

Comment:
I would advocate to ONLY have callable support and move grouping from C to Python ... I don't know if this is the method to do it, but either way it's not for this PR. However, maybe we can discuss that elsewhere not here.

Reply:
I have removed this comment